### PR TITLE
Chroe: add CI pipeline to release docker images to ghcr registry

### PR DESCRIPTION
**Pull request by JaredTan95**

### CODE REVIEW

File: src/api-server/cmd/BUILD.bazel [hunk 1]
```diff
@@ -70,7 +70,8 @@ container_push(
     # aws ecr-public get-login-password --region us-east-1 |\
     #     docker login --username AWS --password-stdin public.ecr.aws/tricorder
     # aws ecr-public create-repository --repository-name api-server
-    registry = "public.ecr.aws/tricorder",
+    # Use bazel run --define=REGISTRY=ghcr.io/tricorder-observability to set this value.
+    registry = "$(REGISTRY)",
```
Comment (nascentcore-eng) [inline src/api-server/cmd/BUILD.bazel:74]: You need to update the build_and_push_images.sh script

**Review by nascentcore-eng — COMMENTED** [review]
I have reverted change to agent/cmd/BUILD.bazel, that can be done in the next step

File: .github/workflows/release-dev-images.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY_SERVER_ADDRESS: ghcr.io/tricorder-observability
+  REGISTRY_USER_NAME: ${{ github.repository_owner }}
+  REGISTRY_PASSWORD: ${{ secrets.REGISTRY_TOKEN }}
+
+jobs:
```
Comment (nascentcore-eng) [inline .github/workflows/release-dev-images.yaml:16]: Need to make this depends on the other job in the build_and_test.yaml
https://github.com/tricorder-observability/starship/blob/e51b91ee35803dab33a72499f380fab054bb8fbb/.github/workflows/build-and-test.yml#L32

Comment (nascentcore-eng) [reply .github/workflows/release-dev-images.yaml:16]: https://stackoverflow.com/a/64733705

**Review by nascentcore-eng — CHANGES_REQUESTED** [review]

File: .github/workflows/release-dev-images.yaml (hunk 2 above)
Comment (JaredTan95) [reply .github/workflows/release-dev-images.yaml:16]: make sense indeed, I will update it. only release images after `build and test` passed.

**Pull request closed without merging**